constraints-dev.txt

```diff
@@ -16,18 +16,18 @@ colorama==0.4.6           # via tox
 contourpy==1.3.2          # via matplotlib
 coverage==7.9.2           # via pytest-cov
 cycler==0.12.1            # via matplotlib
-datasets==3.6.0           # via -r requirements.txt
+datasets==4.0.0           # via -r requirements.txt
```

Comment:
It looks like datasets 4.0.0 is problematic based on these unit test failures, so we should not upgrade to this version.